design/apps.go

```diff
@@ -0,0 +1,260 @@
+	simpleSpace,
+	nil)
+
+var simpleAppSingle = JSONSingle(
+	"SimpleApplication", "Holds a single response to a space/application request",
+	simpleApp,
+	nil)
+
+var simpleEnvironmentSingle = JSONSingle(
+	"SimpleEnvironment", "Holds a single response to a space/environment request",
+	simpleEnvironment,
+	nil)
+
+var simpleEnvironmentMultiple = JSONList(
+	"SimpleEnvironment", "Holds a response to a space/environment request",
+	simpleEnvironment,
+	nil,
+	nil)
+
+var simplePod = a.Type("SimplePod", func() {
+	a.Description("wrapper for a kubernetes Pod")
+	a.Attribute("pod", d.Any)
+})
+
+var simplePodMultiple = JSONList(
+	"SimplePod", "Holds a list of pods",
+	simplePod,
+	nil,
+	nil)
+
+var simpleDeploymentSingle = JSONSingle(
+	"SimpleDeployment", "Holds a single response to a space/application/deployment request",
+	simpleDeployment,
+	nil)
+
+var simpleDeploymentStatSeriesSingle = JSONSingle(
+	"SimpleDeploymentStatSeries", "HOlds a response to a stat series query",
+	simpleDeploymentStatSeries,
+	nil)
+
+var simpleEnvironmentStatSingle = JSONSingle(
+	"EnvStats", "Holds a single response to a pipeline/stats request",
+	envStats,
+	nil)
+
+var _ = a.Resource("apps", func() {
+	a.BasePath("/apps")
+
+	// An auth token is required to call the auth API to get an OpenShift auth token.
+	a.Security("jwt")
+
+	a.Action("showSpace", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID"),
+		)
+		a.Description("list applications in a space")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+		})
+		a.Response(d.OK, simpleSpaceSingle)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showSpaceApp", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName"),
+		)
+		a.Description("list application")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+		})
+		a.Response(d.OK, simpleAppSingle)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showSpaceAppDeployment", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName/deployments/:deployName"),
+		)
+		a.Description("list pipe element")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+			a.Param("deployName", d.String, "Name of the pipe deployment")
+		})
+		a.Response(d.OK, simpleDeploymentSingle)
+		a.Response(d.NotFound, JSONAPIErrors)
+	})
+
+	a.Action("showDeploymentStats", func() {
+		a.Routing(
+			a.GET("/spaces/:spaceID/applications/:appName/deployments/:deployName/stats"),
+		)
+		a.Description("list deployment statistics")
+		a.Params(func() {
+			a.Param("spaceID", d.UUID, "ID of the space")
+			a.Param("appName", d.String, "Name of the application")
+			a.Param("deployName", d.String, "Name of the deployment")
+		})
+		a.Response(d.OK, simpleDeploymentSingle)
+		a.Response(d.NotFound, JSONAPIErrors)
```

Comment:
same comment as above about access restriction. Also, you should always have `InternalServerError`, too.

Reply:
see above comment